Fix for #577 rename convertDateTimeColsToDummies

NateGarrettHC opened the pull request:
Per #577 I changed the convertDateTimeColsToDummies to be named splitOutDateTimeCols. I updated all references of convertDateTimeColsToDummies to reference the correct new name, and I left the function as a shell with a warning that states it has been renamed to splitOutDateTimeCols.

Aylr commented:
@michaellevy This slipped through the cracks - can you asses it?

michaellevy commented:
Also, this function has a lot of copy-and-pasted code. It would be better if each idea were expressed only once, if you feel like getting into that.

NateGarrettHC commented:
I added the .Deprecated function. I also reduced the redundancy in the code. I added the 'format' parameter to allow people to use this function for different formats of dates. I also added a unit test to resolve issue #28.

michaellevy commented:
Thanks for the responsiveness, @NateGarrettHC. I have a couple additional requests before we merge this in:

- [ ] I'd like a warning if the format doesn't seem to be right. Ideally this would be a check after the conversion that is currently the first line of the body of the function. E.g. in:

```r
 df <- data.frame(dcol = "12/10/99")
splitOutDateTimeCols(df, "dcol", format = "%m-%d-%Y")
```

- [ ] There's still some redundancy in the hour/minute/second calculation. See if you can eliminate all duplicated lines.
- [ ] Get rid of the empty `if` condition at the top of the `else if`s. I think it could just be deleted.
- [ ] Use \code{\link{}} to link to `splitOutDateTimeCols` from the documentation of `convertDateTimeColToDummies`.
- [ ] Return the old "value" and "example" sections to the documentation of `convertDateTimeColToDummies`. You can find them in the R directory of the master branch on Github if you want to just copy them over. 

Thanks for polishing this up!

NateGarrettHC commented:
Hey @michaellevy thanks for the good ideas to make the code better. I was able to do all of these, but I'm not committing them because my fix to the redundancy in the code doesn't seem to be a better fix. Instead of having the repeat code, I have multiple if statements, which I thought was not as efficient as having an if else statement, even with the redundant code. Maybe you could look at this and tell me what you think. I couldn't come up with a better way to reduce the code besides this.
```
if (depth == "h" || depth == "m" || depth == "s") {
    df$hour <- as.POSIXlt(df[[dateTimeCol]])$hour
    
  }
    if (depth == "m" || depth == "s") {
    df$min <- as.POSIXlt(df[[dateTimeCol]])$min
  } 
  if (depth == "s") {
    df$Sec <- as.POSIXlt(df[[dateTimeCol]])$sec
  }
```
Let me know what you think and if you feel the multiple if statements is better or you have a different idea for this fix.

michaellevy commented:
@NateGarrettHC I think what you have works. If you wanted to pare down the repetition further, you could convert h, m, s to an ordinal factor, or just integers (3, 2, 1) and do `if (depth >= 1) ... if (depth >=2 ) ...` Or, you could test for depth after adding each column and if you're at the appropriate depth, return from there. Eg:

```r
df$dayOfWeek <- as.POSIXlt(df[[dateTimeCol]])$wday + 1
if (depth == "d") {
  return(df) 
} else {
  df$hour <- as.POSIXlt(df[[dateTimeCol]])$hour
  if (depth == "h") {
    return(df)
  } else {
    ...
  }
}
```

With that approach, you've got to remove the date column some other way, so maybe break `add_columns` out as its own function that the main function calls.

NateGarrettHC commented:
I'm open to whatever you'd prefer since this is your package. I feel that changing the depth parameter to numbers makes it less self-explanatory, but it does allow for less redundancy. And the add_column function also reduces redundancy, but I don't know if nested if statements are better than multiple if statements. It seems like pros and cons with each option, so I'll make it whatever you'd prefer it to be.

michaellevy commented:
@NateGarrettHC - I agree, tradeoffs to each approach. It's fine as you have it. I think checking the date format is the most important thing here.

NateGarrettHC commented:
@michaellevy I've pushed my changes so this is ready for review.

michaellevy commented:
Thanks @NateGarrettHC I will review. And don't worry about that appveyor failure... it's from the warning that the old function now generates in the examples. I think I can easily take care of it.

michaellevy commented on tests/testthat/test-split-Out-Date-Time-Cols.R:
```diff
@@ -0,0 +1,77 @@
+            expect_identical(df, expecteddf)
+  })
+
+test_that("For a DT column with DD/MM/YYYY format for depth='s', function returns expected
```
Fantastic @NateGarrettHC

michaellevy commented:
@NateGarrettHC this looks great. Thanks for your work on this. I made a couple minor changes to get the CI tests passing. Can you give me push access to your branch so I can get the tests passing before merging? Here's the process: https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/

NateGarrettHC commented:
@michaellevy It looks like that box is already checked for me. Can you not push the edits?

michaellevy commented:
@NateGarrettHC I don't have the workflow for editing a PR from a fork down. I created another PR (#772) with all your commits in it. I'll merge it once it's passing tests. Thanks again!
